src/loaders.py

```diff
@@ -11,47 +11,42 @@
 
 
 class RandomSubsetSampler(Sampler):
-    def __init__(self, data_size, subset_size):
-        super().__init__()
+    def __init__(self, data_size, sample_size):
         self.data_size = data_size
-        self.subset_size = subset_size
+        self.sample_size = sample_size
 
     def __iter__(self):
-        return iter(torch.randperm(self.data_size)[:self.subset_size].long())
+        return iter(torch.randperm(self.data_size)[:self.sample_size].long())
 
     def __len__(self):
-        return self.subset_size
+        return self.sample_size
 
 
 class ImageDetectionDataset(Dataset):
-    def __init__(self, X, y, labels, images_dir, encoder, train_mode):
+    def __init__(self, ids, annotations, annotations_human_labels, images_dir, output_encoder, train_mode):
         super().__init__()
-        self.X = X
-        if y is not None:
-            self.y = y
-            self.labels = labels
-        else:
-            self.y = None
-            self.labels = None
+        self.ids = ids
+        self.annotations = annotations
+        self.annotations_human_labels = annotations_human_labels
         self.images_dir = images_dir
-        self.encoder = encoder
+        self.output_encoder = output_encoder
```

Comment:
@jakubczakon Maybe `target_encoder` will be more understandable, what do you think?

Reply:
@taraspiotr that was my first choice but target encoder (or likelihood encoder) is a technique that deals with categorical data so i switched to this

Reply:
I guess using target encoding for this problem that means a different thing is nothing bad

Reply:
@jakubczakon ok